packages/mui-base/src/TextareaAutosize/TextareaAutosize.tsx

```diff
@@ -164,18 +203,23 @@ const TextareaAutosize = React.forwardRef(function TextareaAutosize(props, ref)
         syncHeightWithFlushSycn();
       }
     });
-    const containerWindow = ownerWindow(inputRef.current);
-    containerWindow.addEventListener('resize', handleResize);
-    let resizeObserver;
-
-    if (typeof ResizeObserver !== 'undefined') {
-      resizeObserver = new ResizeObserver(handleResize);
-      resizeObserver.observe(inputRef.current);
+    let resizeObserver: ResizeObserver;
+    let containerWindow: Window;
+    if (inputRef.current) {
+      containerWindow = ownerWindow(inputRef.current);
+      containerWindow.addEventListener('resize', handleResize);
+
+      if (typeof ResizeObserver !== 'undefined') {
+        resizeObserver = new ResizeObserver(handleResize);
+        resizeObserver.observe(inputRef.current);
+      }
     }
 
     return () => {
       handleResize.clear();
-      containerWindow.removeEventListener('resize', handleResize);
+      if (containerWindow) {
```

Comment:
I would be in favor of removing all the defensive checks. e.g. this one and `if (inputRef.current) {`. As far as I know, these are guaranteed to be defined even if TypeScript doesn't know it.

Reply:
:+1: We haven't seen any bug reports related to not having these checks in the JS version, so it's safe to skip them here as well.

@sai6855 would you mind correcting this? The rest of the PR looks fine to me.

Reply:
done, i've removed defensive checks ,  commit: https://github.com/mui/material-ui/pull/35862/commits/cb83064db27d64ca97a131c68d68fd3451d9e941 . cc @michaldudak